### DISCUSSION TITLE
pyaudio stream failing

### DISCUSSION THREAD

**Discussion by Jank14** [snakers4/silero-vad#503] [Q&A] [answered]
Hi, I am using silero vad from long time. After this new update the streaming api is failing which was working fine before that, is there some change required in that file.Attaching screenshot for the reference.The same code worked fine before 
<img width="1229" alt="Screenshot 2024-07-11 at 12 28 20 PM" src="https://github.com/user-attachments/assets/9f78b322-6d0a-42eb-8585-4bddfe24496b">

**Comment by Jank14** [comment]
IS the required for sample rate of 16000 to have chunk size of 512 ? I want to use on ms basis of each chunk being of 250ms. Is there a chance of that

**Comment by snakers4** [comment, marked as answer]
Hi, the new VAD has a fixed chunk size. Also please see the slight changes in invocation.

If you would like to use the older VAD, it can be loaded via a tag.

**Comment by Jank14** [comment]
> Hi, the new VAD has a fixed chunk size. Also please see the slight changes in invocation.
> 
> If you would like to use the older VAD, it can be loaded via a tag.

Hello thanks for the reply, Would like to check the change in functionalities but till that time would like to use old vad, can u share which tag you are mentioning?

**Comment by snakers4** [comment]
v4 or v3.1
See here https://github.com/snakers4/silero-vad/issues/474

**Reply by Jank14** [reply]
> v4 or v3.1 See here #474

Hey thanks for this. Just wanted to check is there any specific reason for limiting the chunk size. Like in that discussion its mentioned vad doesnt work well with big chunk size (which in my case is of 250 ms =4000 samples with 16000 as sample rate). Is the size of 512 a recommendation for good output?

**Reply by snakers4** [reply]
The original VAD was trained with a 32 / 64 / 96 ms chunk size. Larger chunk size provided better quality. 

For the latest model - due to improvements of the architecture there was no difference, so larger chunks were deprecated. 

Also shorter chunk is sought after - because having high quality detector on short chunk enables faster decisions and some logic can be implemented on top of shorter chunks.

Long chunks like 250ms can be obtained by some form of pooling the results of shorter chunks.

**Reply by Jank14** [reply]
> The original VAD was trained with a 32 / 64 / 96 ms chunk size. Larger chunk size provided better quality.
> 
> For the latest model - due to improvements of the architecture there was no difference, so larger chunks were deprecated.
> 
> Also shorter chunk is sought after - because having high quality detector on short chunk enables faster decisions and some logic can be implemented on top of shorter chunks.
> 
> Long chunks like 250ms can be obtained by some form of pooling the results of shorter chunks.

Okay as in perform inferencing on short chunks and then combine the audio ones together to get 250 ms sort of output right? Thanks a lot for the help

**Reply by snakers4** [reply]
Right, you can just do some sort of post-pooling / averaging / simple algorithm / sliding average window on top
